### ISSUE TITLE
`doom-themes` recipe outdated and not building package correctly

### ISSUE THREAD

**Issue by jimeh** [raxod502/straight.el#913]
I might be doing something stupid, but it looks like the recipe that straight.el uses for [`doom-themes`](https://github.com/doomemacs/themes) is out of date, and I can't seem to get it to update.

I've done `M-x straight-pull-all` and `M-x straight-check-all`, also updated and rebuilt the `melpa` package specifically. Locally on disk the `doom-themes` [melpa recipe](https://github.com/melpa/melpa/blob/5f9e73d6ed472924cf17e602a13fde79140ed148/recipes/doom-themes) file (`straight/repos/melpa/recipes/doom-themes`) is up to date containing the following:

```elisp
(doom-themes
 :fetcher github
 :repo "doomemacs/themes"
 :files (:defaults
         "themes/*.el"
         "themes/*/*.el"
         "extensions/*.el"))
```

But using `M-x straight-get-recipe` I get the following:

```elisp
(doom-themes :type git :flavor melpa :files (:defaults "themes/*.el" "doom-themes-pkg.el") :host github :repo "hlissner/emacs-doom-themes")
```

I'm not too familiar with straight's recipe lookup process, so maybe it's not pulling the `doom-themes` recipe from Melpa, but the `:flavor` value suggests it is. And I might be blind, but I didn't easily find anything to indicate what source a recipe comes from, aside from the `:flavor` thing.

The main issue this causes is that all the `doom-themes-ext-*.el` files/modes are not available, as they were recently moved into the `extensions` top-level directory. So they are not built by straight.el, meaning the extensions aren't available.

For now, I've worked around this issue using `straight-recipe-overrides`:

```elisp
(setq straight-recipe-overrides
      '((nil . ((doom-themes :type git :host github
                             :repo "doomemacs/themes"
                             :files (:defaults
                                     "themes/*.el"
                                     "themes/*/*.el"
                                     "extensions/*.el"))))))
```

**Comment by progfolio** [comment]
The recipe was updated on MELPA about two weeks ago.

https://github.com/melpa/melpa/pull/7822

Have you tried `M-x straight-pull-recipe-repositories` to update your local copy of the MELPA recipes?

**Comment by jimeh** [comment]
@progfolio it seems it could have a been a me issue. After doing `M-x straight-rebuild-all`, straight.el now has the correct and up to date recipe without the override.

In the past I've always updateed "all the things" with `M-x straight-pull-all` followed by `M-x straight-check-all`. But seems that didn't do the trick this time. I had also tried `M-x straight-pull-package RET melpa` and `M-x straight-rebuild-package RET melpa` without any luck.

I tried rebuild-all because I got an error about a `*.elc` file in the `with-editor` package, and rebuilding the package fixed it. So I figured there might be others that `straight-check-all` had missed too.

I'm not sure if there's an actual issue here, so I'll leave this issue open for now.

**Comment by progfolio** [comment]
> M-x straight-rebuild-package RET melpa

This is a no-op because we don't build anything with MELPA. Pulling should update the recipes on disk, though.

The `straight-check-all` probably failed because it only rebuilds packages which have been modified. In this case, the doom-themes repository was not modified, only its MELPA recipe was modified.

I think we can close this issue. If a similar issue crops up or this issue persists, just comment here and we can reopen.

**Comment by jimeh** [comment]
Right, rebuild being a no-op makes sense for the `melpa` package. Though the recipe had been updated on disk (`straight/repos/melpa/recipies/doom-themes`), and `straight-pull-all` had pulled the latest changes from the doom-themes repo (old repo URL redirects to the new one). While `straight-check-all` had rebuilt the package but missed all `extensions/*.el` files, yielding a semi-broken doom-themes package. That's how I noticed this issue in the first place cause the `doom-themes-visual-bell-config` command/func was no longer available after I had updated all packages.

And `straight-get-recipe` kept returning the old recipe across Emacs restarts too. It wasn't until I ran `straight-rebuild-all` that straight started picking up the new recipe, despite it being on disk the whole time under `straight/repos/melpa/recipies/doom-themes`.

Basically, something was very weird, but I can't re-produce it anymore, so ¯\\\_(ツ)\_/¯

Also, thanks for your quick responses @progfolio :)
